src/Packetery/Module/Options/BugReportService.php

```diff
@@ -0,0 +1,211 @@
+		foreach ( $form->getErrors() as $error ) {
+			$this->messageManager->flash_message(
+				(string) $error,
+				MessageManager::TYPE_ERROR,
+				MessageManager::RENDERER_PACKETERY,
+				'plugin-options'
+			);
+		}
+	}
+
+	/**
+	 * @param Container $form
+	 */
+	public function onFormValidate( Container $form ): void {
+		$valuesArray = (array) $form->getValues();
+		$message     = isset( $valuesArray['message'] ) ? (string) $valuesArray['message'] : '';
+
+		if ( $message === '' || trim( $this->wpAdapter->wpStripAllTags( $message ) ) === '' ) {
+			/** @var TextArea $messageControl */
+			$messageControl = $form['message'];
+			$messageControl->addError( (string) $this->wpAdapter->__( 'Message is required.', 'packeta' ) );
+		}
+	}
+
+	private function sendBugReport( string $replyEmail, string $message ): bool {
+		$siteName = $this->wpAdapter->getBlogInfo( 'name', 'raw' );
+		$subject  = sprintf( 'Packeta: Plugin WP - hlášení problémů - %s', $siteName );
+
+		/**
+		 * @var array $headers
+		 * @phpstan-var array{0:string,1:string,2:string} $headers
+		 */
+		$headers = [
+			'Content-Type: text/html; charset=UTF-8',
+			'From: ' . $replyEmail,
+			'Reply-To: ' . $replyEmail,
+		];
+
+		$emailBody = $this->createEmailBody( $replyEmail, $message, $siteName );
+
+		$attachments = $this->createAttachments();
+
+		return $this->wpAdapter->wpMail( $this->supportEmailAddress, $subject, $emailBody, $headers, $attachments );
+	}
+
+	private function createEmailBody( string $replyEmail, string $message, string $siteName ): string {
+		return $this->latteEngine->renderToString(
+			PACKETERY_PLUGIN_DIR . '/template/email/bug-report.latte',
+			[
+				'replyEmail'   => $replyEmail,
+				'message'      => $message,
+				'siteName'     => $siteName,
+				'translations' => [
+					'bugReportTitle' => $this->wpAdapter->__( 'Bug Report from Packeta Plugin', 'packeta' ),
+					'website'        => $this->wpAdapter->__( 'Website:', 'packeta' ),
+					'replyEmail'     => $this->wpAdapter->__( 'Reply Email:', 'packeta' ),
+					'message'        => $this->wpAdapter->__( 'Message:', 'packeta' ),
+					'autoGenerated'  => $this->wpAdapter->__( 'This report was automatically generated by the Packeta WordPress plugin.', 'packeta' ),
+				],
+			]
+		);
+	}
+
+	/**
+	 * @return array<int, string>
+	 */
+	private function createAttachments(): array {
+		$attachments = [];
+		$timestamp   = $this->wpAdapter->currentTime( 'Y-m-d_H-i-s', false );
+		$zipFilename = "logs-{$timestamp}.zip";
+
+		$zip     = new ZipArchive();
+		$zipPath = sys_get_temp_dir() . '/' . $zipFilename;
+
+		if ( $zip->open( $zipPath, ZipArchive::CREATE ) === true ) {
+			$this->addPluginSettingsToZip( $zip );
+			$this->addWooCommerceSystemStatusToZip( $zip );
+			$zip->close();
+
+			if ( file_exists( $zipPath ) ) {
+				$attachments[] = $zipPath;
+			}
+		}
+
+		return $attachments;
+	}
+
+	private function addPluginSettingsToZip( ZipArchive $zip ): void {
```

Comment:
Nevidím důvod to mít jako samostatnou metodu. Použil bych jako jeden řádek v místě volání.